OpenSim/Region/Framework/Scenes/SceneObjectPartInventory.cs

```diff
@@ -695,8 +695,16 @@ private void _ReplaceItem(SceneObjectPart part, UUID itemID, bool allowedDrop, b
         {
             TaskInventoryItem newItem = null;
             bool scriptUpdate = false;
-            int type = m_items[itemID].InvType;
-            if (type == 10) // Script
+
+            // Avoid a race condition where the viewer might send what results in a second _Replace(itemID, null) 
+            // for a delete which is a *duplicate*. This can happen if the user deletes the TaskInventoryItem 
+            // a *second* time before the viewer updates the Contents to remove it. This change just handle 
+            // the itemID no longer being present in the items list.
+            if (!m_items.ContainsKey(itemID))
```

Comment:
On further review, I agree. I was concerned that some of the later code, which still uses the itemID, not a reference, might also throw (and another issue or two), so my goal here wasn't as idealist as to eliminate the race condition completely, just to decrease the rate which seemed to increase recently. I also did _not_ want to add any new locks, or change the chronological order of operations here.  However, it seems that the .Remove is safe to call when the itemID is not in the dictionary, and the only real window remaining if we pull the item reference and use that is the .Add (which might add it twice). I'm going to give this one a second kick at the can to see if I can't improve it more without adding risk.